packages/valibot-form-adapter/src/validator.ts

```diff
@@ -2,22 +2,35 @@ import { safeParse, safeParseAsync } from 'valibot'
 import type { BaseIssue, BaseSchema, BaseSchemaAsync } from 'valibot'
 import type { Validator } from '@tanstack/form-core'
 
-export const valibotValidator = (() => {
-  return {
-    validate({ value }, fn) {
-      if (fn.async) return
-      const result = safeParse(fn, value)
-      if (result.success) return
-      return result.issues.map((i) => i.message).join(', ')
-    },
-    async validateAsync({ value }, fn) {
-      const result = await safeParseAsync(fn, value)
-      if (result.success) return
-      return result.issues.map((i) => i.message).join(', ')
-    },
-  }
-}) as Validator<
-  unknown,
-  | BaseSchema<unknown, unknown, BaseIssue<unknown>>
-  | BaseSchemaAsync<unknown, unknown, BaseIssue<unknown>>
->
+type Params = {
+  transformErrors?: (
+    errors: [BaseIssue<unknown>, ...BaseIssue<unknown>[]],
+  ) => string
```

Comment:
To keep it consistent with the other validators this might be:

```suggestion
  transformErrors?: (
    errors: BaseIssue<unknown>[],
  ) => ValidationError
```